tasks/snyk-code-test.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: snyk-code-test
+spec:
+  description: >-
+    Static code security test with snyk
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  params:
+    - name: SHARED_SECRET
+      default: test-team-snyk
+    - name: ARGS
+      type: string
+      description: extra args needs to append
+      default: "--all-projects --exclude=test*,vendor,deps"
+  volumes:
+    - name: snyk-secret
+      csi:
+        readOnly: true
+        driver: csi.sharedresource.openshift.io
+        volumeAttributes:
+          sharedSecret: $(params.SHARED_SECRET)
+  steps:
+    - name: snyk-code-test
+      image: quay.io/redhat-appstudio/hacbs-test:feature-sast
+      workingDir: $(workspaces.workspace.path)
+      volumeMounts:
+        - name: snyk-secret
+          mountPath: "/etc/secrets"
+          readOnly: true
+      script: |
+        #!/usr/bin/env bash
+        SNYK_TOKEN="$(cat /etc/secrets/snyk_token)"
+        if [[ -z $SNYK_TOKEN ]]; then
+          echo "SNYK_TOKEN is empty and a secret 'test-team-snyk' which includes 'snyk_token' need to be created in test-team namespace"
```

Comment:
I think we want to capture this error in the `HACBS_TEST_OUTPUT` result, like on line 55.